src/storage/storage_profiler.h

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#ifndef MXNET_STORAGE_STORAGE_PROFILER_H_
+#define MXNET_STORAGE_STORAGE_PROFILER_H_
+
+#include <mxnet/storage.h>
+#include <string>
+#include <vector>
+#include "../engine/profiler.h"
+
+namespace mxnet {
+namespace storage {
+
+#if MXNET_USE_PROFILER
+/*!
+ * \brief Storage allocation/deallocation profiling via ProfileCounters
+ */
+class DeviceStorageProfiler {
```

Comment:
Why isn't this in the engine profiler?

Reply:
Profiler was previously very engine-specific.  It no longer uniquely connected directly to engine operations, so I have been kind of wondering where it should be moved to in general.  Here I kept it in a header here because it's pretty independcent/self-contained and profiler.h is getting big.

Reply:
Also, I think this serves as a good example of how you can profile something by just using the profile object classes and not directly interacting with the more complex Profiler class.

Reply:
I was actually thinking this might be a subclass inside of StorageImpl...